yt_astro_analysis/halo_analysis/halo_finding_methods.py

```diff
@@ -111,18 +111,33 @@ def _parse_old_halo_list(data_ds, halo_list):
     halo_properties = { f : (np.zeros(num_halos),unit) \
         for f, unit in zip(new_fields,new_units)}
 
+    save_particles = getattr(halo_list, "save_particles", False)
+    if save_particles:
+        n_particles = np.zeros(num_halos, dtype=np.int32)
+
     # Iterate through the halos pulling out their positions and virial quantities
     # and filling in the properties dictionary
     for i,halo in enumerate(halo_list):
-        halo_properties['particle_identifier'][0][i] = i
+        halo_properties['particle_identifier'][0][i] = halo.id
         halo_properties['particle_mass'][0][i] = halo.virial_mass().in_cgs()
         halo_properties['virial_radius'][0][i] = halo.virial_radius().in_cgs()
 
+        if save_particles:
+            n_particles[i] = halo.indices.size
+
         com = halo.center_of_mass().in_cgs()
         halo_properties['particle_position_x'][0][i] = com[0]
         halo_properties['particle_position_y'][0][i] = com[1]
         halo_properties['particle_position_z'][0][i] = com[2]
 
+    if save_particles:
+        i = 0
+        member_ids = np.empty(n_particles.sum(), dtype=np.int64)
+        for halo in halo_list:
```

Comment:
This is not a criticism, but I have to admit that this feels like the type of thing that there should be a numpy-ism for.

Actually, wait -- could this be done with:

`member_ids = np.concatenate(_['particle_index'] for _ in halo_list)` ?  You could keep the allocation and supply an `out` argument to `conncatenate`.

Reply:
Thanks for the suggestion. To my surprise, it is just as fast, even without the pre-allocation.